app/controllers/application_controller.rb

```diff
@@ -58,8 +57,10 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:pending_invite_email])
   end
 
-  def event_staff?(current_event)
-    current_user && current_event.teammates.where(user_id: current_user.id).any?
+  def event_staff?(event)
+    if event && current_user
+      event.teammates.where(user_id: current_user.id).any?
```

Comment:
I like the cleanup here, but it might be worth a few minutes to get rid of it entirely and use current_user.staff_for?(event). event_staff? is one of a few methods like this on application controller that are marked for cleanup: reviewer? and organizer? are also on their way out.